src/ducks/apps/components/Sections/components/Filters.jsx

```diff
@@ -0,0 +1,93 @@
+/* global cozy */
+import React, { useState, useRef } from 'react'
+import Button from 'cozy-ui/transpiled/react/Buttons'
+import IconButton from 'cozy-ui/transpiled/react/IconButton'
+import ActionMenu, {
+  ActionMenuItem,
+  ActionMenuRadio
+} from 'cozy-ui/transpiled/react/ActionMenu'
+import SettingIcon from 'cozy-ui/transpiled/react/Icons/Setting'
+import Icon from 'cozy-ui/transpiled/react/Icon'
+import useBreakpoints from 'cozy-ui/transpiled/react/hooks/useBreakpoints'
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+
+const { BarRight } = cozy.bar
+
+const Filters = ({ onFilterChange }) => {
+  const anchorRef = useRef()
+  const [menuDisplayed, setMenuDisplayed] = useState(false)
+  const [appUnderMaintenance, setAppUnderMaintenance] = useState(true)
+
+  const toggleMenu = () => {
+    setMenuDisplayed(!menuDisplayed)
+  }
+
+  const hideMenu = () => {
+    setMenuDisplayed(false)
+  }
+
+  const showAppUnderMaintenance = () => {
+    setAppUnderMaintenance(true)
+    onFilterChange({})
+  }
+
+  const hideAppUnderMaintenance = () => {
+    setAppUnderMaintenance(false)
+    onFilterChange({
+      underMaintenance: false
+    })
+  }
```

Comment:
When you switch the maintenance radio, it removes the text filter. 

So for example you type 'Ame', it shows 3 results including 1 in maintenance. But when I click on "Hide apps in maintenance", it removes the text filter. So I see all apps not in maintenance and not the 2 'Ame' apps not in maintenance.

Bug or feature ?

Reply:
Good question. The filters are used via the url parameters and so it reloads all the pages. I'll make a note to investigate this behavior further.